capstone/fitness/src/main/java/com/google/sps/progress/JsonGoalStep.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.progress;
+
+import com.google.sps.fit.*;
+import com.google.sps.fit.Exercise.SetType;
+import com.google.sps.util.*;
+import java.util.*;
+
+/* Json version of progress marker. */
+public class JsonGoalStep {
```

Comment:
Can we add methods in GoalStep for conversion between json and pojo instead of have a new class JsonGoalStep or even better use a lib like Jackson API?

Reply:
I looked into the Jackson API and it seems like it also uses a pojo to act as an intermediate with reading and writing. Also it seems like it has to write to a local file/output stream to produce json objects.

I mostly used the pojo because Gson made conversion really simple to and from the GoalStep (see toJson and setJsonGoalSteps in ProgressModel). 

Also, what kind of methods did you have in mind for the conversion between json and pojo?